services/datamanager/model/model.go

```diff
@@ -0,0 +1,324 @@
+	tlsConfig := config.NewTLSConfig(cfg).Config
+	cloudConfig := cfg.Cloud
+	rpcOpts := []rpc.DialOption{
+		rpc.WithTLSConfig(tlsConfig),
+		rpc.WithEntityCredentials(
+			cloudConfig.ID,
+			rpc.Credentials{
+				Type:    rdkutils.CredentialsTypeRobotSecret,
+				Payload: cloudConfig.Secret,
+			}),
+	}
+
+	conn, err := NewConnection(logger, appAddress, rpcOpts)
+	if err != nil {
+		return nil, err
+	}
+	client := NewClient(conn)
+	return NewManager(logger, cfg.Cloud.ID, client, conn, &http.Client{})
+}
+
+// NewManager returns a new model Manager.
+func NewManager(logger golog.Logger, partID string, client v1.ModelServiceClient,
+	conn rpc.ClientConn, httpClient httpClient,
+) (Manager, error) {
+	_, cancelFunc := context.WithCancel(context.Background())
+	ret := modelManager{
+		conn:       conn,
+		client:     client,
+		logger:     logger,
+		cancelFunc: cancelFunc,
+		partID:     partID,
+		httpClient: httpClient,
+	}
+	return &ret, nil
+}
+
+func (m *modelManager) deploy(ctx context.Context, req *v1.DeployRequest) (*v1.DeployResponse, error) {
+	resp, err := m.client.Deploy(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// Close all resources (goroutines) associated with modelManger.
+func (m *modelManager) Close() {
+	m.cancelFunc()
+	if m.conn != nil {
+		if err := m.conn.Close(); err != nil {
+			m.logger.Errorw("error closing model deploy server connection", "error", err)
+		}
+	}
+}
+
+// DownloadModels handles deploying models into their specified destination.
+func (m *modelManager) DownloadModels(cfg *config.Config, modelsToDeploy []*Model, errorChannel chan error) chan error {
+	modelsToDownload, err := getModelsToDownload(modelsToDeploy)
+	if err != nil {
+		errorChannel <- err
+		return errorChannel
+	}
+	// TODO: DATA-295, delete models in file system that are no longer in the config. If we have no models to download, exit.
+	if len(modelsToDownload) == 0 {
+		return nil
+	}
+	// Stop download of previous models if we're trying to download new ones.
+	if m.cancelFunc != nil {
+		m.cancelFunc()
+	}
+
+	cancelCtx, cancelFn := context.WithCancel(context.Background())
+	m.cancelFunc = cancelFn
+	for _, model := range modelsToDownload {
+		deployRequest := &v1.DeployRequest{
+			Metadata: &v1.DeployMetadata{
+				ModelName: model.Name,
+			},
+		}
+		deployResp, err := m.deploy(cancelCtx, deployRequest)
+		if err != nil {
+			m.logger.Error(err)
+			errorChannel <- err
+		}
+		url := deployResp.Message
+		filePath := filepath.Join(model.Destination, model.Name)
+		err = downloadFile(cancelCtx, m.httpClient, filePath, url, m.logger)
+		if err != nil {
+			m.logger.Error(err)
+			errorChannel <- err
+		}
+		// A download from a GCS signed URL only returns one file.
+		modelFileToUnzipPath := filepath.Join(model.Destination, model.Name+zipExtension)
+		if err = unzipSource(modelFileToUnzipPath, model.Destination); err != nil {
+			m.logger.Error(err)
+			errorChannel <- err
+		}
+	}
+	close(errorChannel)
```

Comment:
This close can happen by the caller, and instead don't return anything here.

Reply:
removed `close(errorChannel)` . now it does not appear anywhere. Is this what you meant?